adjustText/__init__.py

```diff
@@ -107,644 +98,328 @@ def get_bboxes(objs, r=None, expand=(1, 1), ax=None, transform=None):
-
-
-def float_to_tuple(a):
-    try:
-        a = float(a)
-        return (a, a)
-    except TypeError:
-        assert len(a) == 2
-        try:
-            b = float(a[0]), float(a[1])
-        except TypeError:
-            raise TypeError("Force values must be castable to floats")
-        return b
+def explode(coords, static_coords, r=None):
+    N = coords.shape[0]
+    x = coords[:, [0, 1]].mean(axis=1)
+    y = coords[:, [2, 3]].mean(axis=1)
+    points = np.vstack([x, y]).T
+    if static_coords.shape[0] > 0:
+        static_x = np.mean(static_coords[:, [0, 1]], axis=1)
+        static_y = np.mean(static_coords[:, [2, 3]], axis=1)
+        static_centers = np.vstack([static_x, static_y]).T
+        points = np.vstack([points, static_centers])
+    tree = scipy.spatial.KDTree(points)
+    pairs = tree.query_pairs(r, output_type="ndarray")
+    pairs = pairs[pairs[:, 0] < N]
+    pairs = pairs[pairs[:, 0] != pairs[:, 1]]
+    diff = points[pairs[:, 0]] - points[pairs[:, 1]]
+    xshifts = np.bincount(pairs[:, 0], diff[:, 0], minlength=N)
+    yshifts = np.bincount(pairs[:, 0], diff[:, 1], minlength=N)
+    return xshifts, yshifts
 
 
 def adjust_text(
     texts,
     x=None,
     y=None,
-    add_objects=None,
-    ax=None,
-    expand_text: float | tuple[float, float] = (1.05, 1.2),
-    expand_points: float | tuple[float, float] = (1.05, 1.2),
-    expand_objects: float | tuple[float, float] = (1.05, 1.2),
-    expand_align: float | tuple[float, float] = (1.05, 1.2),
-    autoalign="xy",
-    va="center",
-    ha="center",
-    force_text: float | tuple[float, float] = (0.1, 0.25),
-    force_points: float | tuple[float, float] = (0.2, 0.5),
-    force_objects: float | tuple[float, float] = (0.1, 0.25),
-    lim=500,
-    precision=0.01,
-    only_move={"points": "xy", "text": "xy", "objects": "xy"},
-    avoid_text=True,
-    avoid_points=True,
+    objects=None,
     avoid_self=True,
-    save_steps=False,
-    save_prefix="",
-    save_format="png",
-    add_step_numbers=True,
+    force_text: tuple[float, float] = (0.1, 0.2),
+    force_static: tuple[float, float] = (0.05, 0.1),
+    force_pull: tuple[float, float] = (0.01, 0.001),
+    explode_force: tuple[float, float] = (0.01, 0.01),
+    expand: tuple[float, float] = (1.05, 1.1),
+    explode_radius="auto",
+    ensure_inside_axes=True,
+    expand_axes=False,
+    only_move={"text": "xy", "static": "xy", "explode": "xy"},
+    ax=None,
+    min_arrow_len=5,
+    time_lim=0.1,
     *args,
-    **kwargs
+    **kwargs,
 ):
-    """Iteratively adjusts the locations of texts.
-
-    Call adjust_text the very last, after all plotting (especially
-    anything that can change the axes limits) has been done. This is
-    because to move texts the function needs to use the dimensions of
-    the axes, and without knowing the final size of the plots the
-    results will be completely nonsensical, or suboptimal.
+    if ax is None:
+        ax = plt.gca()
+    plt.draw()
```

Comment:
I have not tried this version, but I suggest doing something like in #141 to make sure this does what is expected.